kolibri/plugins/coach/assets/src/views/reports/channel-list-page.vue

```diff
@@ -1,59 +1,84 @@
-        <activity-cell :date="lastActive[row.id]" />
-      </tr>
-    </tbody>
-  </report-table>
+  <div>
+    <div v-if="showRecentOnly" ref="recentHeader">
+      <h1>{{ $tr('recentTitle') }}</h1>
+      <sub>{{ $tr('recentSubHeading') }}</sub>
+    </div>
 
-</template>
+    <report-table :caption="$tr('channelList')">
+      <thead slot="thead">
+        <tr>
+          <header-cell :text="$tr('channels')" align="left"/>
+          <header-cell :text="$tr('lastActivity')" align="left"/>
+        </tr>
+      </thead>
+      <tbody slot="tbody">
+        <template v-for="channel in channelList">
+          <tr v-if="channelIsVisible(lastActive[channel.id])" :key="channel.id">
+            <name-cell :kind="CHANNEL" :title="channel.title" :link="reportLink(channel.id)"/>
+            <activity-cell :date="lastActive[channel.id]"/>
+          </tr>
+        </template>
+      </tbody>
+    </report-table>
+  </div>
 
+</template>
 
 <script>
 
-  const ContentNodeKinds = require('kolibri.coreVue.vuex.constants').ContentNodeKinds;
-  const PageNames = require('../../constants').PageNames;
+  const { ContentNodeKinds } = require('kolibri.coreVue.vuex.constants');
+  const { PageNames } = require('../../constants');
   const orderBy = require('lodash/orderBy');
+  const differenceInDays = require('date-fns/difference_in_days');
 
   module.exports = {
     name: 'channelList',
     $trNameSpace: 'coachRecentPageChannelList',
     $trs: {
+      recentTitle: 'Recent Activity',
+      recentSubHeading: 'Showing recent activity in past 7 days',
       channels: 'Channels',
       channelList: 'Channel list',
       lastActivity: 'Last active',
     },
+    data() {
+      return {
+        currentDateTime: Date.now(),
+      };
+    },
     components: {
       'report-table': require('./report-table'),
       'header-cell': require('./table-cells/header-cell'),
       'name-cell': require('./table-cells/name-cell'),
       'activity-cell': require('./table-cells/activity-cell'),
     },
     computed: {
-      CHANNEL() { return ContentNodeKinds.CHANNEL; },
+      CHANNEL() {
+        return ContentNodeKinds.CHANNEL;
+      },
       channelList() {
-        const orderedLists = {
-          [PageNames.RECENT_CHANNELS]: orderBy(
-            this.channels,
+        const orderByArgs = {
+          [PageNames.RECENT_CHANNELS]: [
             [channel => this.lastActive[channel.id] || '', 'title'],
             ['desc', 'asc']
-          ),
-          [PageNames.TOPIC_CHANNELS]: orderBy(this.channels, ['title']),
-          [PageNames.LEARNER_CHANNELS]: orderBy(this.channels, ['title']),
+          ],
+          [PageNames.TOPIC_CHANNELS]: ['title'],
+          [PageNames.LEARNER_CHANNELS]: ['title'],
         };
-        return orderedLists[this.pageName];
+        return orderBy(this.channels, ...orderByArgs[this.pageName]);
       },
     },
     methods: {
+      channelIsVisible(lastActiveTime) {
```

Comment:
Nice. Going to add "Fixes #1265" to automatically close that issue to the description, if you don't mind.